core/auth/src/main/java/software/amazon/awssdk/auth/signer/internal/BaseEventStreamAsyncAws4Signer.java

```diff
@@ -259,4 +267,35 @@ public void subscribe(Subscriber<? super ByteBuffer> s) {
             return transformedRequestBody.contentLength();
         }
     }
+
+    private static String toDebugString(Message m, boolean truncatePayload) {
+        StringBuilder sb = new StringBuilder("Message = {headers={");
+        Map<String, HeaderValue> headers = m.getHeaders();
+
+        Iterator<Map.Entry<String, HeaderValue>> headersIter = headers.entrySet().iterator();
+
+        while (headersIter.hasNext()) {
```

Comment:
why not do this with some kind of stream thing?

Reply:
I found it easier to use this while working with the string builder. I can change it if it's a blocker for you

Reply:
No not a blocker, just a nit. You can ignore.